README.md

```diff
@@ -195,14 +191,10 @@ The following lines should be used to instantiate a profile client:
 
 ```ruby
 # Provide credentials
-provider = MsRestAzure::ApplicationTokenProvider.new(
-       ENV['AZURE_TENANT_ID'],
-       ENV['AZURE_CLIENT_ID'],
-       ENV['AZURE_CLIENT_SECRET'])
-credentials = MsRest::TokenCredentials.new(provider)
-
 options = {
-  credentials: credentials,
+  tenant_id: ENV['AZURE_TENANT_ID'],
+  client_id: ENV['AZURE_CLIENT_ID'],
+  client_secret: ENV['AZURE_CLIENT_SECRET'],
   subscription_id: ENV['AZURE_SUBSCRIPTION_ID']
 }
 
```

Comment:
would our guidelines from before profile changes not work for initializing credentials, I believe we were doing:
```ruby
 provider = MsRestAzure::ApplicationTokenProvider.new(
        ENV['AZURE_TENANT_ID'],
        ENV['AZURE_CLIENT_ID'],
        ENV['AZURE_CLIENT_SECRET'])
    credentials = MsRest::TokenCredentials.new(provider)
    @client = Azure::ARM::Resources::ResourceManagementClient.new(credentials)
    @client.subscription_id = @subscription_id
```
If so, should we leave them as they were, just updating the namespace for ResourceManagementClient? so we demonstrate the minimum change?

Reply:
BTW, after this PR is done, we'd need to re-update the samples that were updated with the last release too.

Reply:
The detailed explanation would be that we always want the user (of the SDK) to provide us with the tenant_id, client_id, client_secret and subscription_id. There is no escaping that. The credentials is just a derived value from these. The user may or may not provide it. 

But, when you think about it, why would a user want to provide a derived value when he/she is supplying the original values anyway? That is the reason, I am updating the examples in readme and removed the credentials. But, if a user wants to provide it, then he is free to do that.

Reply:
So... Credentials are not a bad thing to provide. This abstraction allows any token provider to be used.